### PR TITLE
Remove squeeze/unsqueeze surrounding matmul

### PR DESCRIPTION
**Pull request by LPanosTT**
This branch along with their corresponding [tt-forge-fe](https://github.com/tenstorrent/tt-forge-fe/pull/338) and [tt-mlir](https://github.com/tenstorrent/tt-mlir/pull/818) branches of the same name contain all the necessary fixes to compile and run resnet50 in its entirety through forge --> MLIR.

- Added a TVM pass to remove squeeze/unsqueeze ops surrounding matmul since ttnn can perform matmul on differently-ranked tensors.
- Also fixed how squeezing is done on dual reduce-mean ops so that the rank of the output matches the framework output.

CC: @nvukobratTT

### CODE REVIEW

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 1]
```diff
@@ -3831,6 +3834,50 @@ def callback(self, pre, post, node_map):
         transpose2 = tvm.relay.transpose(transpose1, axes=[0,2,1]) # (n_kv_heads*brcst_val, head_dim, bs*seqlen)
         return transpose2
 
+class RemoveDenseInputSqueeze(DFPatternCallback):
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:3837]: Can you add brief doc to this class why we're doing this :))

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 2]
```diff
@@ -3970,6 +4017,7 @@ def run_forge_compile_passes(relay_module, params=None, inputs=None, target=None
             PadSpecificBatchMatmulShapes(),
             SimplifyVITOnnxAttention(),
             GQABroadcastReshape(),
+            RemoveDenseInputSqueeze(),
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:4020]: Let's add e2e test to confirm this patter isn't broken in the future: `forge/test/mlir/test_features.py`.

File: python/tvm/relay/op/contrib/forge/forge_passes.py (hunk 1 above)
Comment (LPanosTT) [reply python/tvm/relay/op/contrib/forge/forge_passes.py:3837]: Done

File: python/tvm/relay/op/contrib/forge/forge_passes.py (hunk 2 above)
Comment (LPanosTT) [reply python/tvm/relay/op/contrib/forge/forge_passes.py:4028]: Done

**Review by nvukobratTT — APPROVED** [review]

**Pull request merged**